src/package_report.py

```diff
@@ -89,6 +92,93 @@ def _get_installed_package_versions_and_conda_versions(
     return target_packages_match_spec_out, latest_package_versions_in_upstream
 
 
+def _generate_python_package_size_report_per_image(
+    base_version_dir, target_version_dir, image_config, base_version, target_version
+):
+    print("\n# Python Package Size Report " + "(" + image_config["image_type"].upper() + ")\n")
+    print("\n### Target Image Version: " + str(target_version) + " | Base Image Version: " + str(base_version) + "\n")
+    target_pkg_metadata_file = f'{target_version_dir}/{image_config["package_metadata_filename"]}'
+    base_pkg_metadata_file = (
+        f'{base_version_dir}/{image_config["package_metadata_filename"]}' if base_version_dir else None
+    )
+    if not os.path.exists(target_pkg_metadata_file):
+        raise Exception("No Python package metadata file found for target image, please try re-build the image.")
+    with open(target_pkg_metadata_file) as jsonfile:
+        target_pkg_metadata = json.load(jsonfile)
+    base_pkg_metadata = None
+    base_total_size = None
+    if not base_pkg_metadata_file or not os.path.exists(base_pkg_metadata_file):
+        print("WARNING: No Python package metadata file found for base image, only partial results will be shown.")
+    else:
+        with open(base_pkg_metadata_file) as jsonfile:
+            base_pkg_metadata = json.load(jsonfile)
+        base_total_size = sum(d["size"] for d in base_pkg_metadata.values())
+
+    # Print out the total size change of all Python packages in the image.
+    target_total_size = sum(d["size"] for d in target_pkg_metadata.values())
+    total_size_delta_val = (target_total_size - base_total_size) if base_total_size else None
+    total_size_delta_rel = (total_size_delta_val / base_total_size) if base_total_size else None
+    print("\n## Python Packages Total Size Summary\n")
+    print("Target Version Total Size | Base Version Total Size | Size Change (abs) | Size Change (%)")
+    print("---|---|---|---")
+    print(
+        sizeof_fmt(target_total_size)
+        + "|"
+        + (sizeof_fmt(base_total_size) if base_total_size else "-")
+        + "|"
+        + (sizeof_fmt(total_size_delta_val) if total_size_delta_val else "-")
+        + "|"
+        + (str(round(total_size_delta_rel * 100, 2)) if total_size_delta_rel else "-")
+    )
```

Comment:
Now that there are multiple use cases for generating tables, consider refactoring this to use a table data structure, with an accompanying helper method for formatting, so the logic for creating the table markdown can be defined only once.

Reply:
Good point! I'll refactor the codes by adding a common helper that converts table data architecture to markdown table.